### ISSUE TITLE
Fix Element.apply return type

### ISSUE THREAD

**Issue by mathieuleclaire** [uosis/laminar-web-components#8]
Hi,
I can't find how trigger the ```show()``` function of a [SnackBar component](https://github.com/uosis/laminar-web-components/blob/master/material/src/main/scala/material.scala#L2074) or a Dialog.

Thanks !

**Comment by uosis** [comment]
[Here](https://github.com/raquo/laminar-examples/blob/master/src/main/scala/webcomponents/WebComponentsPage.scala#L86) is an example of how to call a function on web component.

**Comment by mathieuleclaire** [comment]
Well, It is not what I want to do. I would like to trigger a SnackBar element from an action triggered any time (like a button click) and not at the mount time. I don't know how to call the relevant methods from an already built snack bar element. Note that it could be the same with a dialog we want to trigger from a button click.

**Comment by raquo** [comment]
@mathieuleclaire `ctx.thisNode` is the Laminar element, i.e. the same instance as the result of `Slider()` in the linked example (see [Slider.scala](https://github.com/raquo/laminar-examples/blob/master/src/main/scala/webcomponents/material/Slider.scala)). However, `Slider.apply` return type is, incorrectly, a generic `HtmlElement`, whereas it should actually be the more specific `El`.

Same for the `SnackBar` component, its apply method return type should be `El`. Then you'll be able to say something like:

```scala
val snackbar = Snackbar(...)

div(
  "Hello",
  snackbar,  // add a snackbar in the DOM
  button(
    "Show",
    onClick --> (_ => snackbar.ref.show()) // call this whenever
  )
)
```

**Comment by mathieuleclaire** [comment]
Well, when this ```ref``` call is done outside a ```onMountCallback```, it returns a wrong type, so that the ```show()``` method does not exist. Here is the error I get:
```scala
value show is not a member of org.scalajs.dom.html.Element
[error]     snackbar.ref.show()
```

**Comment by raquo** [comment]
Yes, that's exactly what I'm saying.

[Snackbar.apply](https://github.com/uosis/laminar-web-components/blob/master/material/src/main/scala/material.scala#L2184) return type needs to be changed from `HtmlElement` to `El`. Just the type signature, the content of the method is ok. Then it will work.

**Comment by uosis** [comment]
Looks like a bug in the generator - I will change this issue to that. Easy fix [here](https://github.com/uosis/laminar-web-components/blob/master/generator/builder.sc#L244).

**Comment by uosis** [comment]
Fixed by 62c203bff63570d0c516efc95dce3e9727e013ec

**Comment by mathieuleclaire** [comment]
Excellent. Sorry, I was focusing on Laminar, to learn it before fixing issues in this project.
